exporter/api.py

```diff
@@ -22,28 +24,70 @@ def export_resources_to_directory(learning_resources):
         learning_resources (list of learningresources.models.LearningResource):
             LearningResources to export in tarball
     Returns:
-        unicode: Newly created temp directory with files inside of it.
+        (unicode, bool):
+            First item is newly created temp directory with files inside of it.
+            Second item is True if a static asset collision was detected.
     """
-    def sanitize(title):
+    def sanitize(url_name):
         """Sanitize title for use in filename."""
         # Limit filename to 200 characters since limit is 256
         # and we have a number and extension too.
-        return slugify(title)[:200]
+        return slugify(url_name)[:200]
 
     tempdir = mkdtemp()
+    collision = False
     try:
+        os.mkdir(os.path.join(tempdir, "static"))
         for learning_resource in learning_resources:
-            filename = "{id}_{title}.xml".format(
-                id=learning_resource.id,
-                title=sanitize(learning_resource.title),
-            )
-            with open(os.path.join(tempdir, filename), 'w') as f:
+            if learning_resource.url_name is not None:
+                filename = "{id}_{url_name}.xml".format(
+                    id=learning_resource.id,
+                    url_name=sanitize(learning_resource.url_name),
+                )
+            else:
+                filename = "{id}.xml".format(id=learning_resource.id)
+
+            type_name = learning_resource.learning_resource_type.name
+            try:
+                os.mkdir(os.path.join(tempdir, type_name))
```

Comment:
This mkdir is still in a loop. I recommend keeping track with a set, like this (tested locally):

```python
def mkdir(path):
    print "made directory {0}".format(path)

def save(filename):
    pass


files = (
    "a/one.png",
    "a/two.png",
    "a/three.png",
    "a/four.png",
    "a/five.png",
    "a/six.png",
    "a/seven.png",
    "b/one.png",
    "b/two.png",
    "b/three.png",
    "b/four.png",
    "b/five.png",
    "b/six.png",
    "b/seven.png",
)

seen = set()
for filename in files:
    path, name = filename.split("/")
    if path not in seen:
        mkdir(path)
        seen.add(path)
    save(file)
```